gems/smithy/spec/interfaces/client/waiters_spec.rb

```diff
@@ -0,0 +1,933 @@
+# frozen_string_literal: true
+
+require_relative '../../spec_helper'
+
+describe 'Client: Waiters' do
```

Comment:
I think this test file should simply test that the wait_until method exists and works, but not necessarily dive into the behavior of the waiters. I think that's best tested for a spec file in smithy client since we are technically testing the abstract poller/waiter classes. You would have to break up the test model to be something like how other plugins are tested there - with each test set adding their own waiter acceptor as needed.